content/blog/adopting-existing-cloud-resources-into-pulumi/index.md

```diff
@@ -0,0 +1,242 @@
+ +   ├─ azure:core:ResourceGroup        main              create
+ +   ├─ azure:network:VirtualNetwork    main              create
+ +   ├─ azure:network:Subnet            internal          create
+ +   ├─ azure:network:NetworkInterface  main              create
+ +   └─ azure:compute:VirtualMachine    main              create
+
+Resources:
+    + 6 to create
+```
+
+Instead of creating new resources though, we want to adopt the existing ones.
+
+#### 3.Add `import` IDs
+
+As we saw earlier, to import existing resources, we simply add `import: <id>` to the resource options of a resource defined in Pulumi.  That resources wil then be imported instead of created.
+
+We can try this for one resource first.  We can look in our `terraform.tfstate` file or in the Azure Portal and see that the id of the Resource Group is `/subscriptions/0282681f-7a9e-424b-80b2-96babd57a8a1/resourceGroups/tfvmex-resources`.  So let's use that:
+
+```ts
+const mainResourceGroup = new azure.core.ResourceGroup("main", {
+    location: "West US 2",
+    name: `${prefix}-resources`,
+}, { import: "/subscriptions/0282681f-7a9e-424b-80b2-96babd57a8a1/resourceGroups/tfvmex-resources" });
+```
+
+#### 4. Adopt resources
+
+Now our preview shows that it will import the `ResourceGroup` instead:
+
+```
+$ pulumi up
+Previewing update (dev2):
+
+     Type                         Name              Plan       
+ +   pulumi:pulumi:Stack          importazure-dev2  create     
+ =   └─ azure:core:ResourceGroup  main              import     
+ 
+Resources:
+    + 1 to create
+    = 1 to import
+    2 changes
+```
+
+We can complete this process, adding `import: <id>` to all resources, and then import them all to Pulumi. 
+
+```
+Updating (dev2):
+
+     Type                               Name              Status       
+ +   pulumi:pulumi:Stack                importazure-dev2  created      
+ =   ├─ azure:core:ResourceGroup        main              imported     
+ =   ├─ azure:network:VirtualNetwork    main              imported     
+ =   ├─ azure:network:Subnet            internal          imported     
+ =   ├─ azure:network:NetworkInterface  main              imported     
+ =   └─ azure:compute:VirtualMachine    main              imported     
+ 
+Resources:
+    + 1 created
+    = 5 imported
+    6 changes
+
+Duration: 8s
+```
+
+#### 5. Managing the resources with Pulumi
+
+Once all the resources are adopted into Pulumi, we can now manage them by making incremental changes in Pulumi, and can discontinue using the existing `.tf` files to manage this set of infrastructure.  For example, if we want to add a new tag to our Virtual Machine, we can add it as below.  We can also now remove the `import` property if we want, or else leave it intact so long as the `id` of the resource remains the same (if in the future we need to make a change that replaces one of these imported resources, Pulumi will require that the `id` proeprty is removed, as it will no longer be correct). 
```

Comment:
```suggestion
Once all of the resources are adopted into Pulumi, we can manage them by making incremental changes in Pulumi, and can discontinue using the existing `.tf` files to manage this set of infrastructure.  For example, if we want to add a new tag to our Virtual Machine, we can use the code below.  We can also now remove the `import` property if we want, or else leave it intact so long as the `id` of the resource remains the same (if in the future we need to make a change that replaces one of these imported resources, Pulumi will require that the `id` property is removed, as it will no longer be correct.) 
```